src/main/kotlin/lotto/domain/LottoPrize.kt

```diff
@@ -1,15 +1,15 @@
 package lotto.domain
 
-enum class LottoPrize(val matchCount: Int, val prizeMoney: Int, val text: String) {
-    MISS(0, 0, "0개 일치"),
-    FIFTH(3, 5_000, "3개 일치"),
-    FOURTH(4, 50_000, "4개 일치"),
-    THIRD(5, 1_500_000, "5개 일치"),
-    SECOND(5, 30_000_000, "5개 일치, 보너스 볼 일치"),
-    FIRST(6, 2_000_000_000, "6개 일치");
+enum class LottoPrize(val matchCount: Int, val prizeMoney: Int, val isMatchBonus: Boolean, val text: String) {
+    MISS(0, 0, false,  "0개 일치"),
+    FIFTH(3, 5_000, false, "3개 일치"),
+    FOURTH(4, 50_000, false,"4개 일치"),
+    THIRD(5, 1_500_000, false, "5개 일치"),
+    SECOND(5, 30_000_000, true,"5개 일치, 보너스 볼 일치"),
+    FIRST(6, 2_000_000_000, false,"6개 일치");
 
     companion object {
-        fun from(matchCount: Int) = values().find { it.matchCount == matchCount } ?: MISS
+        fun from(matchCount: Int, matchBonus : Boolean) = values().find { it.matchCount == matchCount && it.isMatchBonus == matchBonus } ?: MISS
```

Comment:
2,3등 구분이 이제 가능해진 것 같네요! ㅎㅎ 

여기서 한걸음 더 나아가면, 지금은 정적 팩토리 함수에서 매치카운트와 보너스 번호를 모두 직접 판단하는 로직을 가지고 있어요.  근데 이 역시 상위 객체가 하위객체에 대한 조건들을 직접 계산하는 것으로, 보너스번호가 아니라 다른 조건들이 추가된다면 유지보수가 힘들어질 수 있습니다. 

그래서 열거타입의 속성이 람다식을 가지도록 해서 직접 판단하도록 할 수도 있을 것 같아요. 그럼 다음과 같이 추상화가 가능할 것 같아요. 

```suggestion
        fun from(matchCount: Int, matchBonus : Boolean) = values().find { it.support(matchCount, matchBonus) } ?: MISS
```